### ISSUE TITLE
See faction relationships & have a purpose for enemies

### ISSUE THREAD

**Issue by ickerio** [ickerio/factions#51]
### Is your feature request related to a problem?

Currently its impossible to know who you are allied, neutral and an enemy of and enemies have no purpose or functionality.

### Describe the solution you'd like.

The definition of an ally, neutral or enemy is when two factions have mutually declared the same status on each other. The info command should list all factions *mutual* allies, neutrals and enemies of said faction.

I'm not exactly sure how enemies should work but here are some ideas
- Getting killed by a mutual enemy could lose 2x the power on death
- Enemies could do 1.5x damage to each other
- Enemies could be highlighted to the player using the highlight effect when in close proximity (maybe??)

### Describe alternatives you've considered.

No alternatives. High priority features

### Other

Should have a little write up on the wiki for how allies work etc

**Comment by BlueZeeKing** [comment]
We could use enemies as a way to balance out declaring war. If two factions are enemies and one is trying to break the blocks in the other claims, this can increase an "aggression level" between the factions. When this level is high enough, the faction being "attacked" can declare war, opening all claims by the two factions. It will also immediately give the other faction the option to also declare war. We could also disable respawn if you are killed by the other faction or in an enemy chunk during wartime. The war ends when either one side dies enough or they both agree to a peace treaty. That can then be recorded to a sort of "history" of the server and each faction.

**Comment by ickerio** [comment]
Yeah this is really cool. I feel like the aggression level should be based on kills rather than trying to mine unbreakable blocks. Alternatively we could just rename "Enemy" status to "Warring" status and you'd be in a war when you mutually declare `/f declare war <faction>`. We'd also have to put players of the warring factions who join mid war in spectator or adventure mode, the same state as a dead faction member.

**Comment by BlueZeeKing** [comment]
I think we should have both a war and enemy mode. For example in actual history, during the cold war, the US and the USSR were enemies but they weren't directly at war.

**Comment by PyPylia** [comment]
I believe in the scope of this mod, the distinction between warring and enemies is unnecessary. However, I don't think war should need to be declared mutually, only end mutually. This would allow either faction to declare a war instantly, but require both to agree to end it, which I believe would be a better system.

**Comment by BlueZeeKing** [comment]
The first half of this is complete with #72
